### PR TITLE
feat(torch_tensorrt.dynamo.tools): Tool to calculate coverage of PyTorch

### PR DESCRIPTION
**Pull request by narendasan**
opsets

# Description

Compares the Decomposition and Converter Registries to the PyTorch op set to determine coverage. 

Dumps a json file with an op by op breakdown for `ATen` and `prims` when running 

`python -m torch_tensorrt.dynamo.tools.opset_coverage`

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/converter_registry.py [hunk 1]
```diff
@@ -305,23 +305,32 @@ def unique_targets(self):
...
-
+    def get_converter_support_info(self) -> Dict[str, Dict[str, str]]:
+        """Returns a dictionary of targets backed by at least one converter"""
+        available_converters = {}
         for target in sorted(
             self.unique_targets(), key=lambda target: self.qualified_name_or_str(target)
         ):
             _, registry_data = self.get_all_converters_with_target(
                 target, return_registry_info=True
             )
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/converter_registry.py:324]: The `registry_data` dictionary returned by `get_all_converters_with_target` is `Dict[str, int]`, mapping the registry name to the number of instances of converters for that target in each registry. The function header for the output should be updated to `Dict[str, Dict[str, int]]`

File: py/torch_tensorrt/dynamo/tools/opset_coverage.py
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/tools/opset_coverage.py:1]: For evaluation-style operators like `operator.add`, `operator.getitem`, etc. which do not appear in `native_functions.yaml`, but do appear in the ATen operations produced by AOT, could we manually add these to `get_aten_ops`? As in:

```python
...
# Inside get_aten_ops
for op_name, op in {'_operator.floordiv': None,
                    '_operator.mul': None,
                    '_operator.add': None,
                    '_operator.sub': None,
                    '_operator.getitem': None}:
    op_schema_pairs.append((op_name, "Any"))
...

**Review by gs-olive — COMMENTED** [review]
Tool looks great. Added some small suggestions for improvements.

File: py/torch_tensorrt/dynamo/tools/opset_coverage.py
Comment (narendasan) [reply py/torch_tensorrt/dynamo/tools/opset_coverage.py:1]: Do you have a comprehensive list of these ops somewhere?

File: py/torch_tensorrt/dynamo/tools/opset_coverage.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+import torch
+import torch._prims as prims
+import torchgen
+from torch._ops import OpOverload
+from torch_tensorrt.dynamo.conversion.converter_registry import (
+    DYNAMO_CONVERTERS,
+    ConverterRegistry,
+)
+from torch_tensorrt.dynamo.lowering._decompositions import DECOMPOSITIONS
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/tools/opset_coverage.py:18]: `DECOMPOSITIONS` should be accessed through the getter function `get_decompositions`, within the same file

File: py/torch_tensorrt/dynamo/tools/opset_coverage.py
Comment (gs-olive) [reply py/torch_tensorrt/dynamo/tools/opset_coverage.py:1]: I have seen these two lists: [list 1 (omits getitem)](https://github.com/pytorch/pytorch/blob/ce608712cbbb797cb8ea12c4dc6ed3c8dce6d71e/torch/fx/experimental/normalize.py#L126-L141), [list 2](https://github.com/pytorch/pytorch/blob/ce608712cbbb797cb8ea12c4dc6ed3c8dce6d71e/torch/_dynamo/variables/builtin.py#L123-L153)

**Review by gs-olive — APPROVED** [review]
The tool looks great! An idea for a future improvement/feature is, once it's available, to gauge the coverage on the [core ATen IR](https://pytorch.org/docs/stable/ir.html#core-aten-ir) and report stats for this.

**Comment by narendasan** [comment]
Yup want to make a CI job and a board to list whats covered and what isnt
